optimum/intel/openvino/modeling_decoder.py

```diff
@@ -248,6 +251,59 @@ def compile(self):
 class OVModelForCausalLM(OVBaseDecoderModel, GenerationMixin):
     export_feature = "text-generation"
     auto_model_class = AutoModelForCausalLM
+    
+    def __init__(self,
+                 model: openvino.runtime.Model,
+                 config: PretrainedConfig = None,
+                 device: str = "CPU",
+                 dynamic_shapes: bool = True,
+                 ov_config: Dict[str, str] | None = None,
+                 model_save_dir: str | Path | TemporaryDirectory | None = None,
+                 **kwargs):
+        model = self._try_modify_model_io_to_bf16(model, ov_config, device, **kwargs)
+        super().__init__(model, config, device, dynamic_shapes, ov_config, model_save_dir, **kwargs)
+
+    def _try_modify_model_io_to_bf16(self,
+                                     model: openvino.runtime.Model,
+                                     ov_config: Dict[str, str] | None = None,
+                                     device: str = "CPU",
+                                     **kwargs):
+        if device == 'CPU':
+            # if INFERENCE_PRECISION_HINT in ov_config equals to bf16, pastkv will use bf16
+            pastkv_will_use = Type.bf16 if ov_config and ov_config.get("INFERENCE_PRECISION_HINT", "") == "bf16" else Type.f32
+
+            # for batch testsing can set INFERENCE_PRECISION_HINT in enviroment, pastkv will try to use the specific precision
+            if "INFERENCE_PRECISION_HINT" in os.environ:
```

Comment:
Can we use the INFERENCE_PRECISION_HINT from CompiledModel.get_property("INFERENCE_PRECISION_HINT") instead? This would use BF16 by default on CPUs that support it and FP32 on CPUs that don't. You should not need to configure inference precision to get the best performance on SPR, and using the compiled model's inference precision would be more robust towards future API changes too.

It would also be best if the solution is generic enough that it's easy to adapt for other devices (it seem possible that this is relevant for GPU inference too, where inference precision is FP16 by default).

Reply:
Good idea, use core.get_property("INFERENCE_PRECISION_HINT"). If path is still under 'cpu' path, I'm not sure it works for other plugins.